types/passport/passport-tests.ts

```diff
@@ -38,20 +38,27 @@ passport.framework(newFramework);
 interface TestUser {
     id: number;
 }
-passport.serializeUser((user: TestUser, done: (err: any, id?: number) => void) => {
+passport.serializeUser<TestUser, number>((user: TestUser, done: (err: any, id?: number) => void) => {
```

Comment:
I'm pretty nervous to merge a breaking change to a popular package.

I think you can write this with overloads in the .d.ts file with the higher-arity function first and avoid requiring type parameters at all existing call sites.

Reply:
I tried overloading once; but the issue is that both `serialize` functions take in only 1 argument, which is a function.

This function argument has different parameter numbers. If I were to use overloads, I would get a linter error stating that I need to combine the two different types.

Reply:
Example:

```        serializeUser<TUser, TID>(fn: SerializeFunctionWithRequest<TUser, TID>): void;
        serializeUser<TUser, TID>(fn: SerializeFunction<TUser, TID>): void;```

This makes tslint complain